app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,13 +11,21 @@ def self.call(*args)
       end
 
       def call
-        # This method should find the user based on the `receiver_id`.
-        # It should then determine the appropriate Broadcast for a user,
-        # based on the `created_at` and the different conditions for sending a notification.
-        # `welcome_broadcast = ...`
+        return if commented_on_welcome_thread? || received_notification?
 
-        # Once it has the appropriate Broadcast to be sent, it should send a notification for it:
-        # `Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)`
+        welcome_broadcast = Broadcast.find_by(title: "Welcome Notification: welcome_thread")
+        Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)
+      end
+
+      def received_notification?
+        welcome_broadcast = Broadcast.find_by(title: "Welcome Notification: welcome_thread")
```

Comment:
We're making 2 calls to the database by finding the same broadcast here on line 21, and on line 16. Instead, we could pull them out into a (private) method and memoize it so that we make only one call to the database and then use the memoized value :)

```
def welcome_broadcast
  @welcome_broadcast ||= Broadcast.find_by(title: "Welcome Notification: welcome_thread")
end
```

Does that make sense? Happy to explain what's happening here if it is confusing at all!